frontend/tests/utils/wmgUtils.ts

```diff
@@ -88,10 +94,33 @@ export const checkSourceData = async (page: Page) => {
   //click on source data icon
   await page.locator('[data-testid="source-data-button"]').click();
 
-  // number of elemet displayed on source data
+  // number of element displayed on source data
   const n = await page.locator('[data-testid="source-data-list"] a').count();
+
   // close the pop-up
-  await page.keyboard.press("Escape");
```

Comment:
sometimes pressing escape wasn't closing the side panel, so wrapping this to retry and assert the panel is indeed closed. This was causing flaky failure because somehow it's hard delete selected filter when the side panel is still open